ios/ZingoMobile/AppDelegate.m

```diff
@@ -244,4 +260,152 @@ -(void)scheduleProcessingTask {
         // No fallback
     }
 }
+
+// NEW BACKGROUND SCHEDULING TASKS
+
+- (void)handleBackgroundTask {
+    // We require the background task to run when connected to the power and wifi
+    Reachability *reachability = [Reachability reachabilityForInternetConnection];
+    NetworkStatus networkStatus = [reachability currentReachabilityStatus];
+    
+    if (networkStatus == ReachableViaWiFi) {
+      isConnectedToWifi = true;
+    } else {
+      isConnectedToWifi = false;
+    }
+    NSLog(@"BGTask isConnectedToWifi %@", isConnectedToWifi ? @"true" : @"false");
+    
+    [self registerTasks];
+}
+
+- (void)registerTasks {
+    BOOL bcgSyncTaskResult;
+    bcgSyncTaskResult = [[BGTaskScheduler sharedScheduler] registerForTaskWithIdentifier:syncTask usingQueue:dispatch_get_main_queue() launchHandler:^(BGTask * _Nonnull task) {
+        if (![task isKindOfClass:[BGProcessingTask class]]) {
+            return;
+        }
+        NSLog(@"BGTask BGTaskScheduler.shared.register SYNC called");
+        [self startBackgroundTask:(BGProcessingTask *)task];
+    }];
+    
+    NSLog(@"BGTask SYNC registered %d", bcgSyncTaskResult);
+
+    BOOL bcgSchedulerTaskResult;
+    bcgSchedulerTaskResult = [[BGTaskScheduler sharedScheduler] registerForTaskWithIdentifier:syncSchedulerTask usingQueue:dispatch_get_main_queue() launchHandler:^(BGTask * _Nonnull task) {
+        if (![task isKindOfClass:[BGProcessingTask class]]) {
+            return;
+        }
+        NSLog(@"BGTask BGTaskScheduler.shared.register SCHEDULER called");
+        [self scheduleSchedulerBackgroundTask];
+        [self scheduleBackgroundTask];
+        [task setTaskCompletedWithSuccess:YES];
+    }];
+    
+    NSLog(@"BGTask SCHEDULER registered %d", bcgSchedulerTaskResult);
+}
+
+
+- (void)startBackgroundTask:(BGProcessingTask *)task {
+    NSLog(@"BGTask startBackgroundTask called");
+    
+    // Schedule tasks for the next time
+    [self scheduleBackgroundTask];
+    [self scheduleSchedulerBackgroundTask];
+    
+    if (!isConnectedToWifi) {
+        NSLog(@"BGTask startBackgroundTask: not connected to the wifi");
+        [task setTaskCompletedWithSuccess:NO];
+        return;
+    }
+    
+    // Start the syncing
+    NSLog(@"BGTask configureProcessingTask run");
+    syncFinished = false;
+    [NSThread detachNewThreadSelector:@selector(syncingProcessBackgroundTask:) toTarget:self withObject:nil];
+    [self syncingStatusProcessBackgroundTask:nil];
+    [task setTaskCompletedWithSuccess:YES];
```

Comment:
This line must be deleted, your goal is to start the syncing and leave it until expirationHandler is called.